ReactiveCocoa/Swift/Flatten.swift

```diff
@@ -459,20 +459,56 @@ private final class ConcatState<Value, Error: Swift.Error> {
 
 	/// Subscribes to the given signal producer.
 	func startNextSignalProducer(_ signalProducer: SignalProducer<Value, Error>) {
```

Comment:
It seems like this should be modified to not take any arguments since it's going to start _n_ producers. It can check whether the `queue` is empty. This should also simplify the code in `enqueueSignalProducer` above.